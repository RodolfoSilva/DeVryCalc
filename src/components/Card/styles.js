import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderColor: '#EEE',
    backgroundColor: '#FFF',
    borderWidth: 1,
    marginBottom: 15,
    borderRadius: 3
  },
  wrapper: {
    padding: 15,
  }
});

export default styles;
