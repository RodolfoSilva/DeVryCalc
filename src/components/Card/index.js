import React, { PureComponent } from 'react'
import { View, TouchableOpacity } from 'react-native'
import styles from './styles'
import PropTypes from 'prop-types'

class Card extends PureComponent {
  renderCard() {
    return (
      <View style={styles.container}>
        <View style={styles.wrapper}>{this.props.children}</View>
      </View>
    )
  }

  render() {
    if (!this.props.onClick) {
      return this.renderCard()
    }

    return (
      <TouchableOpacity onPress={() => this.props.onClick()}>
        {this.renderCard()}
      </TouchableOpacity>
    )
  }
}

Card.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired
}

export default Card
