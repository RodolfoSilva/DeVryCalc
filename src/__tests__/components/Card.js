import 'react-native'
import React from 'react'
import Card from '../../components/Card'
import { shallow } from 'enzyme'

describe('Card', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Card children="Oi" />)
    expect(wrapper).toMatchSnapshot()
  })
})
