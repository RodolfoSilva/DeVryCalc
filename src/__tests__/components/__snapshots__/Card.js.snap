// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderColor": "#EEE",
      "borderRadius": 3,
      "borderWidth": 1,
      "flex": 1,
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "padding": 15,
      }
    }
  >
    Oi
  </View>
</View>
`;
